framework/include/utils/MooseEnum.h

```diff
@@ -38,6 +38,16 @@ class MooseEnum : public MooseEnumBase
 {
 public:
   /**
+   * Enum item for controlling comparison in the compareCurrent method.
+   */
+  enum CompareMode
```

Comment:
Minor: I like the new C++11 enum class syntax if you aren't using these as array indices (which you aren't).